Use BLST portable feature to enable runtime ADX instruction set check (#1630)
input-output-hk/mithril: mithril-stm/CHANGELOG.md

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

> ## 0.3.18 (TBD-04-2024)

Reviewer: ```suggestion
## 0.3.18 (11-04-2024)
```

> - Deprecate `portable` feature, BLST runtime check of intel ADX instruction set is now enabled. The portable feature now
>  do nothing and should be removed from your crate dependencies.

Reviewer: ```suggestion
- Deprecate `portable` feature, BLST runtime check of intel ADX instruction set is now enabled. The portable feature now has no effect and should be removed from crate dependencies..
```
